Created a README.md file (#10)
neo4j/graph-schema-json-js-utils: README.md

# Graph Schema JSON util.

> ## Consume

Reviewer: Here you should add how the user can install the utility package as a dependency.

```bash
npm install @neo4j/graph-schema-utils
```

> The contents of this repot allow you to validate a graph schema to see if it follows the correct graph specification. The purpose is to allow you to extract the relationships that exist in the database. The result is displayed in the form of a JSON schema, which should also be able to be translated into a .js object and then back into a JSON schema.

Reviewer: There are two parts to this repo:
1. A JSON schema that describes how a graph schema should be expressed in JSON.
2. Utility functions that help the consumers to validate and work with a graph schema.

## Use
json-schema.json
The file contains of the description against which the graph schema will be validated. The JSON schema serves as a template or as the structure that the graph schema is expected to have.

## The validation 
This function is needed to perform validation on a graph schema. The validateSchema function compares the output against the JSON schema. 
```js
export {validateSchema, SchemaValidationError};

> ```

Reviewer: Show how to import (consume), not how to export.

Author: I'm a little unsure about this comment, but make sure I've got it right.

The SchemaValidationError class captures the error message that occurs if an error is detected during validation.
```js
export function validateSchema(jsonSchema, graphSchema) 
```

## Parser

> To make it easy to work with the schema.

Reviewer: ```suggestion
Since the references in the JSON document are references by id:s, there's a parser utility that hydrates the references and makes it easy to work with the schema.
```

```js

> const parsed = model.GraphSchemaRepresentation.parseJson(myModel);

Reviewer: Show how to import the `model` package as well.

```

## Serializer

> To make it easier to save the schema or to send it to someone else.

Reviewer: ```suggestion
If you need to transport or persist the schema, you can serialize the model.
```

```js
const serialized = myModel.toJson();
```

## Test
Unit testing uses the Vitest framework to get fast results. The package consists of tests divided into two groups.
One group 'model' tests the structure of the graph schema when it is parsed and so that it is parsed correctly. 
The second group 'validation' tests the validation on the different levels of the JSON schema to see if the validation works correctly. 

> The package also consists of the file 'fs.utils.js'. It is a function that reads the file to be validated. 

Reviewer: I think it's enough to describe how to run the tests, the technology behind the tests (and the structure) might change.

# Graph Schema JSON util.

## Install
```
npm install @neo4j/graph-schema-utils
```

## Consume
There are two parts to this repo:

1. A JSON schema that describes how a graph schema should be expressed in JSON.
2. Utility functions that help the consumers to validate and work with a graph schema. 

## Use
json-schema.json
The file contains of the description against which the graph schema will be validated. The JSON schema serves as a template or as the structure that the graph schema is expected to have.

## The validation 
This function is needed to perform a validation on a graph schema. The validateSchema function compares the output against the JSON schema. 
```js
import { validateSchema } from "@neo4j/graph-schema-utils";

> ```
> 
> The SchemaValidationError class captures the error message that occurs if an error is detected during validation.
> ```js
> export function validateSchema(jsonSchema, graphSchema) 

Reviewer: ```suggestion
validateSchema(jsonSchema, graphSchema) 
```

# Graph Schema JSON util.

## Install
```
npm install @neo4j/graph-schema-utils
```

## Consume
There are two parts to this repo:

1. A JSON schema that describes how a graph schema should be expressed in JSON.
2. Utility functions that help the consumers to validate and work with a graph schema. 

### JSON-schema

> The JSON that describes the shape of a graph schema is available in the source code (`packages/json-schema/json-schema.js`) and published on: https://dist.neo4j.org/json-graph-schema/

Reviewer: ```suggestion
The JSON that describes the shape of a graph schema is available in the source code (`packages/json-schema/json-schema.json`) and published on: https://dist.neo4j.org/json-graph-schema/
```

## Use
json-schema.json
The file contains of the description against which the graph schema will be validated. The JSON schema serves as a template or as the structure that the graph schema is expected to have.

## The validation 
This function is needed to perform a validation on a graph schema. The validateSchema function compares the output against the JSON schema. 
```js

> import { SchemaValidationError, validateSchema } from "../../src/index.js";

Reviewer: I think it's very rare to need to error class.
And the import should be as the consumer would import it, using the package name.

```suggestion
import { validateSchema } from "@neo4j/graph-schema-utils";
```

```

The SchemaValidationError class captures the error message that occurs if an error is detected during validation.
```js
export function validateSchema(jsonSchema, graphSchema) 
```

## Parser
Since the references in the JSON document are references by id:s, there's a parser utility that hydrates the references and makes it easy to work with the schema.
```js

> import { model } from "../../src";

Reviewer: Same as above, the import would come from the package, not the file system.

> const parsed = model.GraphSchemaRepresentation.parseJson(myModel);

Reviewer: ```suggestion
const parsed = model.GraphSchemaRepresentation.parseJson(graphSchemaJsonString);
```

```

## Serializer
If you need to transport or persist the schema, you can serialize the model.
```js
const serialized = myModel.toJson();
```

## Run test
```npm run test ``` 

> Run ```npm run test:watch``` to let the test watch the users file changes live.

Reviewer: This part renders strange in the document. For inline code, use single \`, for code blocks use triple \`\`\`.

# Graph Schema JSON util.

## Install
```
npm install @neo4j/graph-schema-utils
```

## Consume
There are two parts to this repo:

1. A JSON schema that describes how a graph schema should be expressed in JSON.
2. Utility functions that help the consumers to validate and work with a graph schema. 

> This allows the user to validate a graph schema to see if it follows the correct graph specification. The result is displayed in the form of a JSON schema, which should also be able to be translated into a .js object and into a JSON string.

Reviewer: Maybe delete these two sentences - as they are described below.

> ## Use

Reviewer: ```suggestion
### JSON-schema
```

> json-schema.json
> The file contains of the description against which the graph schema will be validated. The JSON schema serves as a template or as the structure that the graph schema is expected to have.

Reviewer: ```suggestion
The JSON that describes the shape of a graph schema is available in the source code (`packages/json-schema/json-schema.js`) and published on: https://dist.neo4j.org/json-graph-schema/
```

Use this schema to validate graph schemas.

> ## The validation 

Reviewer: ```suggestion
### Validate
```

This function is needed to perform a validation on a graph schema. The validateSchema function compares the output against the JSON schema. 
```js
import { validateSchema } from "@neo4j/graph-schema-utils";
```

The SchemaValidationError class captures the error message that occurs if an error is detected during validation.
```js
export function validateSchema(jsonSchema, graphSchema) 
```

> ## Parser

Reviewer: ```suggestion
### Parse
```

Since the references in the JSON document are references by id:s, there's a parser utility that hydrates the references and makes it easy to work with the schema.
```js
import { model } from "@neo4j/graph-schema-utils";

const parsed = model.GraphSchemaRepresentation.parseJson(graphSchemaJsonString);
```

> ## Serializer

Reviewer: ```suggestion
### Serialize
```

If you need to transport or persist the schema, you can serialize the model.
```js
const serialized = myModel.toJson();
```

> ## Run test

Reviewer: ```suggestion
## Contribute
We welcome contributions to this repo. Fork this repo and open a PR!

### Run test
```

> ```
> npm run test 
> ``` 
> Run `npm run test:watch` to let the test watch the users file changes live.

Reviewer: ````suggestion
To run a single test run:

```bash
npm run test 
``` 

To watch for changes and run tests on change:

```bash
npm run test:watch
```

````
